tests/Kiota.Builder.Tests/Extensions/OpenApiUrlTreeNodeExtensionsTests.cs

```diff
@@ -1072,6 +1072,20 @@ public void repro4085()
         Assert.Equal("\\path\\{differentThingId-id}", differentThingId.Path);
         Assert.Equal("{+baseurl}/path/{differentThingId%2Did}", differentThingId.GetUrlTemplate());
     }
+
+    [Theory]
+    [InlineData("{path}", "WithPath")]
+    [InlineData("archived{path}", "archivedWithPath")]
+    [InlineData("files{path}", "filesWithPath")]
+    [InlineData("name(idParam='{id}')", "nameWithId")]
+    [InlineData("name(idParam={id})", "nameWithId")]
```

Comment:
```suggestion
    [InlineData("name(idParam={id})", "nameWithId")]
    [InlineData("name(idParamFoo={id})", "nameWithId")]
```

This second one will end up colliding with the current logic even though they are entirely different path segments and valid from a service perspective.

Reply:
I was intentionally avoiding this as the current behavior involves pulling only the indexer section `{id}` for the indexer name. To disambiguate further by pulling the `idParamFoo` or `idParam`(parameter name) would break scenario scenarios such as functions in graph. e.g. `delta(token=''{token2}'')` would change from DeltaWithToken to something like `DeltaWithTokenToken2`. In most scenarios, the param name also matches the placeholder and would maybe be `DeltaWithTokenToken`

In my head, this PR would not be changing the handling of functions presence of `(` and `)` characters (the parameter name and its placeholder typically match up) but really the indexers that exist in a path segment that have different prefixes similar to how Kiota handles different suffixes for known file types present in indexers with 
 https://github.com/microsoft/kiota/blob/3987aaee1603c36a7693353c21e4211d4d06ec66/src/Kiota.Builder/Extensions/OpenApiUrlTreeNodeExtensions.cs#L180

Reply:
The challenge with that is we're leaking OData concerns into kiota, which doesn't seem right.
But I think that specific collision risk is fairly constrained to OData services.
And that specific scenario is invalid according to OData. (overloads with the same parameters)
I'd be ok ruling that PR as an improvement over the current situation, and leaving a comment here/in the implementation about that specific concern for future reference in case we ever run into it.